projects/aca-folder-rules/src/lib/services/folder-rule-sets.service.ts

```diff
@@ -0,0 +1,170 @@
+  folderInfo$: Observable<NodeInfo> = this.folderInfoSource.asObservable();
+  isLoading$ = this.isLoadingSource.asObservable();
+
+  constructor(private apiService: AlfrescoApiService, private contentApi: ContentApiService, private folderRulesService: FolderRulesService) {}
+
+  private callApi(path: string, httpMethod: string, body: object = {}): Promise<any> {
+    // APIs used by this service are still private and not yet available for public use
+    const params = [{}, {}, {}, {}, body, ['application/json'], ['application/json']];
+    return this.apiService.getInstance().contentPrivateClient.callApi(path, httpMethod, ...params);
+  }
+
+  private getRuleSets(nodeId: string, skipCount = 0): Observable<RuleSet[]> {
+    return from(
+      this.callApi(
+        `/nodes/${nodeId}/rule-sets?include=isLinkedTo,owningFolder,linkedToBy&skipCount=${skipCount}&maxItems=${FolderRuleSetsService.MAX_RULE_SETS_PER_GET}`,
+        'GET'
+      )
+    ).pipe(
+      tap((res) => {
+        if (res?.list?.pagination) {
+          this.hasMoreRuleSets = res.list.pagination.hasMoreItems;
+        }
+      }),
+      switchMap((res) => this.formatRuleSets(res))
+    );
+  }
+
+  loadRuleSets(nodeId: string) {
+    this.isLoadingSource.next(true);
+    this.ruleSets = [];
+    this.hasMoreRuleSets = true;
+    this.currentFolder = null;
+    this.ruleSetListingSource.next(this.ruleSets);
+    this.hasMoreRuleSetsSource.next(this.hasMoreRuleSets);
+    this.getNodeInfo(nodeId)
+      .pipe(
+        tap((nodeInfo: NodeInfo) => {
+          this.currentFolder = nodeInfo;
+          this.folderInfoSource.next(this.currentFolder);
+        }),
+        switchMap(() => this.getRuleSets(nodeId)),
+        finalize(() => this.isLoadingSource.next(false))
+      )
+      .subscribe((ruleSets: RuleSet[]) => {
+        this.ruleSets = ruleSets;
+        this.ruleSetListingSource.next(this.ruleSets);
+        this.hasMoreRuleSetsSource.next(this.hasMoreRuleSets);
+        this.folderRulesService.selectRule(this.getOwnedOrLinkedRuleSet()?.rules[0] ?? ruleSets[0]?.rules[0]);
+      });
+  }
+
+  loadMoreRuleSets(selectLastRule = false) {
+    this.isLoadingSource.next(true);
+    this.getRuleSets(this.currentFolder.id, this.ruleSets.length)
+      .pipe(finalize(() => this.isLoadingSource.next(false)))
+      .subscribe((ruleSets) => {
+        this.ruleSets.push(...ruleSets);
+        this.ruleSetListingSource.next(this.ruleSets);
+        this.hasMoreRuleSetsSource.next(this.hasMoreRuleSets);
+        if (selectLastRule) {
+          const ownedRuleSet = this.getOwnedOrLinkedRuleSet();
+          this.folderRulesService.selectRule(ownedRuleSet?.rules[ownedRuleSet.rules.length - 1]);
+        }
+      });
+  }
+
+  private getNodeInfo(nodeId: string): Observable<NodeInfo> {
+    if (nodeId) {
+      return this.contentApi.getNode(nodeId).pipe(
+        catchError((error) => {
+          if (error.status === 404) {
```

Comment:
Do we have some file with constants that we use across whole app? If not it may be worth to consider creating one for example for storing HTTP status codes as constants instead of using just raw numbers

Reply:
As far as I know, no, but HTTP codes are pretty well known so perhaps it's not as crucial here.